digitalocean/resource_digitalocean_monitor_alert.go

```diff
@@ -0,0 +1,291 @@
+		DeleteContext: resourceDigitalOceanMonitorAlertDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"uuid": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Required: true,
+			},
+
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					godo.DropletCPUUtilizationPercent,
+					godo.DropletMemoryUtilizationPercent,
+					godo.DropletDiskUtilizationPercent,
+					godo.DropletPublicOutboundBandwidthRate,
+					godo.DropletDiskReadRate,
+					godo.DropletDiskWriteRate,
+					godo.DropletOneMinuteLoadAverage,
+					godo.DropletFiveMinuteLoadAverage,
+					godo.DropletFifteenMinuteLoadAverage,
+					// these are available as constants ...
+					"v1/insights/droplet/public_inbound_bandwidth",
+					"v1/insights/droplet/private_outbound_bandwidth",
+					"v1/insights/droplet/private_inbound_bandwidth",
+				}, false),
+			},
+
+			"compare": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(godo.GreaterThan),
+					string(godo.LessThan),
+				}, false),
+				Description: "The comparison operator to use for value",
+			},
+
+			"description": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Description of the alert policy",
+			},
+
+			"enabled": {
+				Type:     schema.TypeBool,
+				Default:  true,
+				Required: true,
+			},
+
+			"value": {
+				Type:         schema.TypeFloat,
+				Required:     true,
+				ValidateFunc: validation.FloatAtLeast(0),
+			},
+
+			"tags": tagsSchema(),
+
+			"alerts": {
+				Type:        schema.TypeList,
+				Computed:    false,
+				Required:    false,
+				Description: "List with details how to notify about the alert. Support for Slack or email.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"slack": {
+							Type:     schema.TypeList,
+							Required: false,
+							Computed: false,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"channel": {
+										Type:         schema.TypeList,
+										Computed:     false,
+										Required:     true,
+										Description:  "The Slack channel to send alerts to",
+										ValidateFunc: validation.StringIsNotEmpty,
+									},
+									"url": {
+										Type:             schema.TypeString,
+										Computed:         false,
+										DiffSuppressFunc: CaseSensitive,
+										Description:      "The webhook URL for Slack",
+										ValidateFunc:     validation.StringIsNotEmpty,
+									},
+								},
+							},
+						},
+						"email": {
+							Type:         schema.TypeList,
+							Computed:     false,
+							Required:     false,
+							ValidateFunc: validation.ListOfUniqueStrings,
```

Comment:
```suggestion
							Type:     schema.TypeList,
							Computed: false,
							Optional: true,
							Elem: &schema.Schema{
								Type: schema.TypeString,
							},
```

`TypeList` requires an `Elem` defining what the items in the list should be.

https://www.terraform.io/docs/extend/schemas/schema-types.html#typelist

Reply:
Thanks :)